app/controllers/admin/stats_controller.rb

```diff
@@ -29,10 +29,23 @@ def top_depositors_count
     end
 
     def document_by_permission
+      return document_by_date_by_permission if @users_stats[:file_start_date]
+
       files_count = {}
       files_count[:total] = GenericFile.count
-      files_count[:public] = GenericFile.where(Solrizer.solr_name('read_access_group', :symbol) => 'public').count
-      files_count[:registered] = GenericFile.where(Solrizer.solr_name('read_access_group', :symbol) =>'registered').count
+      files_count[:public] = GenericFile.where_public.count
+      files_count[:registered] = GenericFile.where_registered.count
+      files_count[:private] = files_count[:total] - (files_count[:registered] + files_count[:public])
+      files_count
+    end
+
+    def document_by_date_by_permission
+      start_date = DateTime.parse @users_stats[:file_start_date]
+      end_date = DateTime.parse(@users_stats[:file_end_date]).end_of_day unless  @users_stats[:file_end_date].blank?
+      files_count = {}
+      files_count[:total] = GenericFile.find_by_date_created(start_date, end_date).count
```

Comment:
Instead of calling `GenericFile.find_by_date_created(start_date, end_date)` three times on lines 46-48, call it once (and assign it to a variable or put it behind a private method).